From faf48c3ad68083920e0bc7343acd1516f32d115b Mon Sep 17 00:00:00 2001
From: Richard Zhu <hongxing.zhu@nxp.com>
Date: Tue, 16 Oct 2018 12:31:51 +0800
Subject: [PATCH 4863/5242] MLK-19932 PCI: imx: make sure the ref_clk is
 enabled

commit  c0078a8c21ed3520605b5be9280ea786f511767d from
https://source.codeaurora.org/external/imx/linux-imx.git

In the EP RC validation system, EP wouldn't set RC's
CLK_REQ# to low, like one real inserted EP device to do.
Let RC to set the over ride low and enabled, make sure that
REF_CLK of RC side is turned on.

Signed-off-by: Richard Zhu <hongxing.zhu@nxp.com>
Reviewed-by: Fugang Duan <fugang.duan@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/pci/controller/dwc/pci-imx6.c |   34 +++++++++++++++++----------------
 1 file changed, 18 insertions(+), 16 deletions(-)

diff --git a/drivers/pci/controller/dwc/pci-imx6.c b/drivers/pci/controller/dwc/pci-imx6.c
index d9a9bbe..eca22c7 100644
--- a/drivers/pci/controller/dwc/pci-imx6.c
+++ b/drivers/pci/controller/dwc/pci-imx6.c
@@ -1637,25 +1637,27 @@ static int imx_pcie_host_init(struct pcie_port *pp)
 		ret = imx_pcie_establish_link(imx_pcie);
 		if (ret < 0)
 			return ret;
-		/*
-		 * Disable the over ride after link up.
-		 * Let the the CLK_REQ# controlled by HW L1SS
-		 * automatically.
-		 */
-		switch (imx_pcie->variant) {
-		case IMX8MQ:
-		case IMX8MM:
-			if (imx_pcie->ctrl_id == 0)
-				val = IOMUXC_GPR14;
-			else
-				val = IOMUXC_GPR16;
+		if (!IS_ENABLED(CONFIG_RC_MODE_IN_EP_RC_SYS)) {
+			/*
+			 * Disable the over ride after link up.
+			 * Let the the CLK_REQ# controlled by HW L1SS
+			 * automatically.
+			 */
+			switch (imx_pcie->variant) {
+			case IMX8MQ:
+			case IMX8MM:
+				if (imx_pcie->ctrl_id == 0)
+					val = IOMUXC_GPR14;
+				else
+					val = IOMUXC_GPR16;
 
-			regmap_update_bits(imx_pcie->iomuxc_gpr, val,
+				regmap_update_bits(imx_pcie->iomuxc_gpr, val,
 					IMX8MQ_GPR_PCIE_CLK_REQ_OVERRIDE_EN,
 					0);
-			break;
-		default:
-			break;
+				break;
+			default:
+				break;
+			}
 		}
 
 		if (IS_ENABLED(CONFIG_PCI_MSI))
-- 
1.7.9.5

